cppcache/src/CacheableObjectArray.cpp

```diff
@@ -53,6 +54,18 @@ size_t CacheableObjectArray::objectSize() const {
   }
   return size;
 }
+
+std::string CacheableObjectArray::getClassName() const {
+  if (!empty()) {
+    auto&& item = *begin();
+
+    if (auto pdx = dynamic_cast<PdxSerializable*>(item.get())) {
+      return pdx->getClassName();
+    }
+  }
+
+  return "java.lang.Object";
```

Comment:
What if we make this string a `const` value in this compilation unit and change the return type of this method to return `const std::string&` so we aren't copy constructing strings on return.